### DISCUSSION TITLE
clustering_pairwise_predictions_at_threshold

### DISCUSSION THREAD

**Discussion by JLtran11** [moj-analytical-services/splink#2432] [General]
Hello,

I have a question about the output of the cluster_pairwise_preds_at_threshold function. Does the returned DataFrame include IDs from the input table that were not part of any candidate pairs? For example, if we have records a, b, and c, where only a and b form a candidate pair due to blocking rules, will the output look like this:

(record a, cluster 1)
(record b, cluster 1)
(record c, cluster 2)

Or do I need to identify IDs not in candidate pairs from the original table and append them manually to the result?

Additionally, regarding the linker.predict function, which has the argument materialise_blocked_pairs=True, how can I access the materialized blocked pairs table? I had a look around the docs, but could not find anything in relationship to either question.

Thank you!

**Comment by RobinL** [comment]
Yes the output is guaranteed to be the same length as the input nodes, even if some nodes have no edges (links)

On (2) I'm afraid you can't!  It's materialised but them immediately deleted after use.  

You could use https://moj-analytical-services.github.io/splink/api_docs/inference.html#splink.internals.linker_components.inference.LinkerInference.deterministic_link to get a similar result
